llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -4382,6 +4382,38 @@ KnownBits SelectionDAG::computeKnownBits(SDValue Op, const APInt &DemandedElts,
     Known.Zero |= APInt::getBitsSetFrom(BitWidth, VT.getScalarSizeInBits());
     break;
   }
+  case ISD::ATOMIC_LOAD: {
+    // If we are looking at the loaded value.
+    if (Op.getResNo() == 0) {
```

Comment:
(minor) Possibly could be an assertion, chain result is not a part of data flow

Reply:
I was keeping consistency with ISD::LOAD. Maybe we should change all of them at once?